src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -29,39 +29,34 @@ internal sealed partial class AssetProvider(Checksum solutionChecksum, SolutionA
     private readonly SolutionAssetCache _assetCache = assetCache;
     private readonly IAssetSource _assetSource = assetSource;
 
-    private T GetRequiredAsset<T>(Checksum checksum)
```

Comment:
this operation si fundamentally not supported.  there is no way, ever, to ensure that a checksum you're asking for is actually in the asset provider (unles sit was pinned.  but that's a vanishingly small number of cases).